src/data/finset/basic.lean

```diff
@@ -1362,6 +1362,18 @@ by simp only [mem_def, image_val, mem_erase_dup, multiset.mem_map, exists_prop]
 theorem mem_image_of_mem (f : α → β) {a} {s : finset α} (h : a ∈ s) : f a ∈ s.image f :=
 mem_image.2 ⟨_, h, rfl⟩
 
+lemma filter_mem_image_eq_image (f : α → β) (s : finset α) (t : finset β) (h : ∀ x ∈ s, f x ∈ t) :
```

Comment:
Also, would it be more or less useful to state `h` as `t ⊇ s.image f`?

Reply:
I originally had it as a simp lemma, but I think it's never applicable because of that `h` hypothesis.

It's possible to state it using `t ⊇ s.image f`, but it seems like this is a fact that's not too easy to use directly except through `image_subset_iff`.  (The precedent I was following was the `hf` argument to `card_le_card_of_inj_on`.)